application/ui/src/features/inspect/toolbar/sources/source-factory.component.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState } from 'react';
+
+import { isEmpty } from 'lodash-es';
+
+import { $api } from '../../../../api/client';
+import { EditSourceFactory } from './edit-source-factory.component';
+import { SourcesList } from './source-list/source-list.component';
+import { SourceOptions } from './source-options.component';
+import { SourceConfig } from './util';
+
+export const SourceFactory = () => {
+    const sourcesQuery = $api.useSuspenseQuery('get', '/api/sources');
+    const sources = sourcesQuery.data ?? [];
+    const [view, setView] = useState<'list' | 'options' | 'edit'>(isEmpty(sources) ? 'options' : 'list');
```

Comment:
Should `options` be part of this state? I.e. can it not always be derived from `sources` being empty?